Code of Conduct.md

```diff
@@ -0,0 +1,78 @@
+  - Ask for permission before directly messaging another member.
+  - Not use any contact details shared in the group for unsolicited contact or commercial purposes.
+
+## 3. **Event and Speaker Etiquette**
+
+- Give everyone a fair opportunity to contribute to discussions.
+- Respect speakers’ wishes regarding questions—whether they prefer questions throughout their talk or at the end.
+- Constructive criticism is welcome, but only when relevant to the discussion.
+- Feedback on talks is not given unless requested.
+
+## 4. **Alcohol and Social Conduct**
+
+- Many of our events take place in pubs where alcohol is available.
+- Excessive drunkenness will result in a warning and, if repeated, further action.
+- Drinking is entirely optional, and no one should feel pressured to consume alcohol.
+- Peer pressure around alcohol consumption is not acceptable.
+- Members are welcome to buy drinks for others, but should always ask first before purchasing alcohol for someone else.
+- The organisers will not be responsible for tracking informal drink-buying arrangements between members, such as buying rounds.
+
+## 5. **Venue Conduct**
+
+- Most of our meetups take place in public spaces. Members should be mindful of our impact on that environment. Members should be aware of their impact on the environment, including keeping noise levels appropriate and avoiding inconsiderate behaviour such as blocking walkways.
+- Many venues host us for free, so we should:
+  - Treat staff with courtesy and respect.
+  - Ensure we do not create additional work for them (e.g., keeping the space tidy).
+
+## 6. **Promotion and Commercial Interests**
+
+- Members may not use the meetup to promote commercial services or recruit unless responding to a direct request (e.g., someone mentions they are looking for work).
+- Commercial selling or advertising is not allowed.
+- It is acceptable to recommend a service if it is relevant to the discussion and has been asked about.
+- Asking for help with non-commercial projects is fine, but repeated requests should be avoided if there is no interest.
+
+## 7. **Photography and Recording**
+
+- Organisers may take photos for the website, Meetup, or social media.
+- Members will always be asked before a photo is taken and may opt out.
+- If members wish to take photos, they must ensure everyone in the picture has given consent.
+
+## 8. Reporting and Enforcement
+
+- Any concerns should be reported directly to the organisers in person, via WhatsApp, or Meetup.
+- The organisers will handle all disputes privately, and their decisions are final.
+- The organisers will keep records of all reports to ensure consistency in enforcement and to refine the Code of Conduct where necessary.
+
+### **Breaches and Consequences**
+
+- Minor breaches (e.g., an inappropriate joke) will typically result in a warning.
+- Serious breaches (e.g., overt racism, harassment) will result in immediate expulsion at the organisers' discretion.

```

Comment:
I suppose the "organisers' discretion" is necessary to cover issues that aren't clean cut. But I don't like that it sounds there could be wiggle room for overt racism.